cli/src/semgrep/app/scans.py

```diff
@@ -48,17 +50,17 @@ class ScanCompleteResult:
 
 
 class ScanHandler:
```

Comment:
It would be great to remove this intermediate class actually. I'd rather build an actual object
when we received the scan_config ATD record from the backend.

Reply:
Right now we init() very early, with no info, and are forced to setup many fields as Optional and None, because the actual data we get in fetch_and_init_scan_config(), so I'd rather make this fetcn_and_init the init function.

Reply:
Yeah, I agree this is a good change to make - we should do this separately though, the size of this PR is already getting pretty large just from the test refactoring, and I worry that doing both in one change will be even harder to review.